File: prova2/exercicios/cesar.c
#include <stdio.h>
#include <string.h>

char* cifra_cesar(char *text, int N, int deslocamento){
    for (int i = 0; i < N; i++){
        
        if (*(text + i) == ' '){
            continue;
        }
        int ascii_val = (int)*(text + i);
        int posicao_alfabeto = ascii_val - (int)'a'; // Ex: b - a = 1 -> (indice 1 do alfabeto)

        int new_index = posicao_alfabeto - deslocamento;

        int normalized = new_index % 26;
        if(normalized < 0) normalized += 26; // Em python não precisa dessa operação

        *(text + i) = (char)normalized + (int)'a';
    }

    return text;
}
    

void main(){
    char text[] = "hzn oerir erivfnb fboer pevcgbtensvn";

    printf("%s", cifra_cesar(text, strlen(text), 13));

}

File: prova2/exercicios/cesar.py
def shift_letter(text: str, offset: int):
    result = ''
    for letter in text:
        if letter == ' ':
            result += ' '
            continue

        # Usar valor ASCII
        ascii_val = ord(letter)
        current_index = ascii_val - ord('A')
        new_index = current_index - offset

        normalized = new_index % 26
        result += chr(normalized + ord('A'))

    return result


text = "Hzn oerir erivfnb fboer pevcgbtensvn".upper()

for i in range(26):
    result = shift_letter(text, i)
    print(f"Shift {i}: ", result)
